### ISSUE TITLE
Upgrade Dependencies to mitigate CVEs

### ISSUE THREAD

**Issue by mr1716** [apache/accumulo#2891]
**Describe the bug**
Several Dependencies have open CVEs, with the most recent accumulo version. These new dependency versions should remove these vulnerabilities!

Software:
com.google.code.gson » gson1 | 2.8.5 | 2.9.1
com.google.guava » guava1 | 28.0-jre | 31.1-jre|
com.google.protobuf » protobuf-java1 |3.7.1	|3.21.5|
commons-io » commons-io| 2.6 | 2.11.0|
log4j » log4j | 1.2.17 | 2.18.0|
org.apache.thrift » libthrift | 0.12.0 | 0.16.0|

**Comment by EdColeman** [comment]
Some of the versions that you listed have already been updated on the main branch (0e571cafc4) - for example thrift is already 0.16.0

**Comment by mr1716** [comment]
@EdColeman Which ones have been updated and will these updates be incorporated into the next release?

**Comment by dlmarion** [comment]
@mr1716 - which branch / version of the parent pom are you looking at for this issue?

**Comment by milleruntime** [comment]
@EdColeman I see you opened a separate issue with a different set of deps in https://github.com/apache/accumulo/pull/2893. Can this one be closed then?

**Comment by mr1716** [comment]
@milleruntime that ticket doesnt address the com.google.guava » guava1 vulnerability. Could we add that to the referenced ticket or create a separate ticket for it?

**Comment by mr1716** [comment]
@EdColeman It would probably make sense to file another/separate pull request to fix the Guava vulnerability.

**Comment by EdColeman** [comment]
The current guava version in the pom (on the main branch) appears to be the latest that is available in the maven repos. In the pom:

From https://github.com/apache/accumulo/blob/a63f06dbc2d575f4cccf08e9129902ddeacac0cf/pom.xml#L295

```
      <dependency>
        <groupId>com.google.guava</groupId>
        <artifactId>guava</artifactId>
        <version>31.1-jre</version>
      </dependency>
```

Is there another dependency that I missed?

**Comment by EdColeman** [comment]
@milleruntime I opened a PR, not an issue - once that is merged this issue could be closed. I think it address deps in this issue that have available updates plus some that were not mentioned.

**Comment by EdColeman** [comment]
When the release from 2.1 is created it will be from the main branch, so yes, once that PR is merged they would be included in the release.  It is likely that another sweep of the dependencies will be done just prior to the release to capture updated versions that might occur between merging the current PR and the when the release is prepared.

**Comment by EdColeman** [comment]
Closed with #2893

**Comment by dlmarion** [comment]
Did a ticket in accumulo-testing get created (based on my comment [here](https://github.com/apache/accumulo/pull/2893#issuecomment-1225884609)) ?

**Comment by EdColeman** [comment]
I'm working it now
